### ISSUE TITLE
Use NCES data for School/Organization on Registration page

### ISSUE THREAD

**Issue by mattpen** [phetsims/website#460]
From: Kathy Perkins
Date: Wed, Jul 13, 2016 at 11:42 AM
Subject: Fwd: Re: Demographic profile - still participating

As we consider what to do with the Profiles and data we ask from teachers, we should consider leveraging this database for US schools from NCES to tie teachers better to exact schools. This Demographic profiling team isn't going to be able to do a great job of matching up with schools because people enter so many different formats. BHS, Boulder HS, Boulder High, Boulder High School 
Kathy 

-------- Forwarded Message --------
Subject:	Re: Demographic profile - still participating
Date:	Wed, 13 Jul 2016 10:38:13 -0700
From:	Aly Sharp 

Hi Kathy,

Yes, NCES is an open data source you could certainly use. From what I understand, even their naming conventions aren't always consistent, but the nice thing about NCES is that each school and district has a unique numerical identifier you could pull along with the other institution information. This makes it much more straightforward to link to demographic/economic/etc. information about each institution. They update their data annually, but it's usually a couple years old by the time it gets released.

**Comment by mattpen** [comment]
I found sources for the NCES data, it appears that the private schools and public schools are separated into different reports.

[Private Schools](https://nces.ed.gov/surveys/pss/pssdata.asp)
[Public Schools](https://nces.ed.gov/ccd/pubschuniv.asp)

**Comment by oliver-phet** [comment]
Along with using the new NCES data, there will be large changes to the registration page(s).

Mockup of new (and old) registration process can be found here: https://drive.google.com/a/colorado.edu/file/d/0B6CMwxdP0NGYOWZQc0NCOUxJTkk/view?usp=sharing

If a user selects 'Teacher' as a user type, they will receive a different 3rd page "Tell us about your classroom".

If a user does not select 'Teacher' as user type, they receive a general 3rd page "Tell us about your organization".

If a user selects *only* 'Student', a popup message is shown notifying them students are not required to create accounts to use the sims, but allows them to proceed with account creation.

New tabbed Account Information designs are also included in the .ai file.

@mattpen There's a lot here. Let me know if you'd like this segmented into separate issues, etc.

**Comment by oliver-phet** [comment]
Also assigning this to @ariel-phet to prioritize.

**Comment by ariel-phet** [comment]
@mattpen I know this is high priority for @kathy-phet, and probably the one large'ish project before the website reskin.

Not sure if this would also be a good candidate for experimenting with a different framework.

**Comment by ariel-phet** [comment]
Also, this would be a perfectly reasonable project to pair with @zepumph on

**Comment by mattpen** [comment]
@ariel-phet - There is a back-log of profile issues that might be good to work on along with this, especially considering we are moving towards production with the salesforce integration project this week.  Should we tackle these too, or keep deferring?

https://github.com/phetsims/website/milestone/5

**Comment by kathy-phet** [comment]
@ariel-phet - Did Oliver ask you to ask Edmodo what they use? We were looking at their registration and it was really nice. Their registration seems to have a database of all schools - internationally. We were wondering if we might be able to get access to that database so we can do this with all schools?

**Comment by oliver-phet** [comment]
@kathy-phet We did ask about that and they are using NCES for US schools and the international schools were added by users. Unfortunately, they were not willing to share the international database with us.

**Comment by kathy-phet** [comment]
What do you think about doing something similar and building our own database up?

**Comment by ariel-phet** [comment]
@kathy-phet and @oliver-phet I imagine one (tedious) thing we could do is log onto edmodo and copy over what they have for international stuff in their database.

**Comment by oliver-phet** [comment]
I was wondering that too. I think it would work well if we implemented it in a similar way to edmodo - with autocomplete suggestions. I'm not sure how difficult something like that is to develop (maybe @mattpen can chime in).

Also note: their database isn't perfect. If I type in 'Boulder' I get "University of Colorado, Boulder" and "University of Colorado at Boulder".

@ariel-phet I'm not seeing a practical/possible way to copy the info from edmodo. I don't see a way to generate a list of the schools other than by starting to type.

**Comment by mattpen** [comment]
> log onto edmodo and copy over what they have for international stuff in their database.

I don't think this would work. The form info isn't stored on the page, its done via web requests. We'd have to reverse engineer their API which is definitely not feasible.

>  I think it would work well if we implemented it in a similar way to edmodo - with autocomplete suggestions. I'm not sure how difficult something like that is to develop.

We already have an autocomplete method for the search bar, it shouldn't be too hard to adapt this for school data.  I'd be more worried about how to validate user input, given the problems with profane user data we've had in the past.

**Comment by ariel-phet** [comment]
@mattpen @oliver-phet building up our own database seems like the way to go then. Agreed on profanity filter, it might make sense to start building this up **after** we have made the changes that are intended to have students less encouraged to make accounts.

In addition, we could have a human review of that database on a weekly basis for awhile. And as we find things, we can add them to the profanity filter.

**Comment by oliver-phet** [comment]
We will prioritize this data integration after the registration pages have been reskinned (#547).

Currently, a lot of students are registering for the site, and the new design should (hopefully) reduce the amount of student registrations.
